FWCore/Framework/src/processEDAliases.cc

```diff
@@ -0,0 +1,222 @@
+              << "One has module label '" << moduleLabel << "' and product instance name '" << productInstanceName
+              << "',\n"
+              << "the other has module label '" << iter->second.moduleLabel() << "' and product instance name '"
+              << iter->second.productInstanceName() << "'.\n";
+        }
+      } else {
+        auto prodIter = preg.productList().find(key);
+        if (prodIter != preg.productList().end()) {
+          if (!prodIter->second.produced()) {
+            throw Exception(errors::Configuration, "EDAlias\n")
+                << "The module label alias '" << alias << "' and product instance alias '" << theInstanceAlias << "'\n"
+                << "are used for a product of type '" << friendlyClassName << "'\n"
+                << "with module label '" << moduleLabel << "' and product instance name '" << productInstanceName
+                << "',\n"
+                << "An EDAlias can only be used for products produced in the current process. This one is not.\n";
+          }
+          aliasMap.insert(std::make_pair(key, aliasKey));
+          aliasKeys.insert(std::make_pair(aliasKey, key));
+        }
+      }
+    }
+  }  // namespace
+
+  namespace detail {
+    void processEDAliases(std::vector<std::string> const& aliasNamesToProcess,
+                          std::unordered_set<std::string> const& aliasModulesToProcess,
+                          ParameterSet const& proc_pset,
+                          std::string const& processName,
+                          ProductRegistry& preg) {
+      if (aliasNamesToProcess.empty()) {
+        return;
+      }
+      std::string const star("*");
+      std::string const empty("");
+      ParameterSetDescription desc;
+      desc.add<std::string>("type");
+      desc.add<std::string>("fromProductInstance", star);
+      desc.add<std::string>("toProductInstance", star);
+
+      std::multimap<BranchKey, BranchKey> aliasMap;
+
+      std::map<BranchKey, BranchKey> aliasKeys;  // Used to search for duplicates or clashes.
+
+      // Auxiliary search structure to support wildcard for friendlyClassName
+      std::multimap<std::string, BranchKey> moduleLabelToBranches;
+      for (auto const& prod : preg.productList()) {
+        if (processName == prod.second.processName()) {
+          moduleLabelToBranches.emplace(prod.first.moduleLabel(), prod.first);
+        }
+      }
+
+      // Now, loop over the alias information and store it in aliasMap.
+      for (std::string const& alias : aliasNamesToProcess) {
+        ParameterSet const& aliasPSet = proc_pset.getParameterSet(alias);
+        std::vector<std::string> vPSetNames = aliasPSet.getParameterNamesForType<VParameterSet>();
+        for (std::string const& moduleLabel : vPSetNames) {
+          if (not aliasModulesToProcess.empty() and
+              aliasModulesToProcess.find(moduleLabel) == aliasModulesToProcess.end()) {
+            continue;
+          }
```

Comment:
I added the `aliasModulesToProcess` argument and these lines when extracting the `processEDAliases()` function into a separate file.